### DISCUSSION TITLE
Open one time chat channel via API / URL link

### DISCUSSION THREAD

**Discussion by lastmetroid** [Chatterino/chatterino2#5409] [Ideas]
Would it be possible to implement some sort of API function to Chatterino where you could open a URL like this for example:

chatterino://joinchannel/channelname

or something like that. And if you send the URL from another app, it opens and joins that channel, but also once, if you close Chatterino the channel is not in the auto connect list anymore.

**Comment by Felanbird** [comment]
You can run the chatterino.exe with `-c channelNameHere` from a terminal and it will open a new Chatterino window with only that channel, that would probably suffice for your use case.

**Comment by lastmetroid** [comment]
Thanks for the hint. I tried it, but it is not an option because it always runs another process everytime you execute the command. Can you maybe please implement another option for one instance please and also an option to limit chat to one chat only. Also there is no option it seems to close the Chat / not save them when you close Chatterino please also implement this.

Example:

Running chatterino.exe -c channel, Chatterino opens and joins that channel. When you now run another chatterino.exe -c channel2 it should NOT run another Chatterino process but switch the channel to channel2. Maybe with another flag something like chatterino.exe --oneinstance --changechannel -c channel

So if you execute several times

chatterino.exe --oneinstance --changechannel -c channel

it detects chatterino is already running and wont open another window/process and changes the current connected channel to channel.

Also:

chatterino.exe --oneinstance -c channel

Would result in one process but with several tabs.

Then there needs an global option in the settings where you can activate that Chatterino wont save the channels, so everytime you re open Chatterino all channels are closed.

**Comment by pajlada** [comment]
Could you describe your use case?

**Comment by lastmetroid** [comment]
Open Twitch Chats via API/URL to have chat function next to video vial MPC or VLC.
